### DISCUSSION TITLE
Best way to integrate MDX in vue but dynamically and remote

### DISCUSSION THREAD

**Discussion by irux** [mdx-js/mdx#2363] [Q&A] [answered]
Hello guys! I would like to integrate MDX in the following way:

I have a SPA written in Vue and I would like to fetch MDX files and render them inside a div or anywhere in my application. The idea would be to allow the MDX use local and remote components outside the SPA project. Is there anyway to achieve this ? I can imagine that from a security perspective this is not Ideal because you have things like xss and so on, but I should be able to trust the content. I just want a way to have custom data rendered with the flexibility MDX can give me :)
Btw, I would like not to migrate my SPA to use JSX/TSX. The idea is to simple download it and somehow inject it (I don't know if you would need to compile it on the front or get it already compiled from a server)

Any ideas how to achieve this ? 

Thanks!

**Comment by wooorm** [comment, marked as answer]
Sounds like https://mdxjs.com/guides/mdx-on-demand/.
vue latest has an automatic jsx runtime, which you can pass, so that’s all you need on the JSX side
